src/persistence/settings.cpp

```diff
@@ -1379,6 +1409,48 @@ void Settings::setCircleExpanded(int id, bool expanded)
     circleLst[id].expanded = expanded;
 }
 
+void Settings::addFriendRequest(const QString &friendAddress, const QString &message)
+{
+    QMutexLocker locker{&bigLock};
+    QPair<QString, QString> request(friendAddress, message);
+
+    if (friendRequests.indexOf(request) != -1)
+        return;
+
+    friendRequests.push_back(request);
+    ++unreadFriendRequests;
+}
+
+unsigned int Settings::getUnreadFriendRequests() const
+{
+    QMutexLocker locker{&bigLock};
```

Comment:
You are correct about the mutex overhead. I don't think you can get rid of them in getters/setters. Different threads may get/set at a time, for instance the audio is in a separate thread. Bad things will happen if they occur simultaneously.